app/services/personal_key_formatter.rb

```diff
@@ -3,14 +3,15 @@ class PersonalKeyFormatter
   WORD_COUNT = IdentityConfig.store.recovery_code_length
   VALID_CHAR = '[a-zA-Z0-9]'.freeze
   VALID_WORD = "#{VALID_CHAR}{#{CHAR_COUNT}}".freeze
-  REGEXP = "(?:#{VALID_WORD}([\s-])?){#{WORD_COUNT - 1}}#{VALID_WORD}".freeze
+  REGEXP_STRING = "(?:#{VALID_WORD}([\\s\\-])?){#{WORD_COUNT - 1}}#{VALID_WORD}".freeze
```

Comment:
I think we kind of have a goofy couple of bugs here where we had `[\s-]` which evaluates to `[ -]` because the `\` needs to be double escaped to show up as a `"\"` and Ruby turned it into a space. The trailing `-` in a character class works in Ruby, but appears to need to be escaped when we pass the string to be a pattern in HTML.

Reply:
I thought we stopped having users re-type personal key? So we can remove this from rendering it in the HTML right?

Reply:
it's also used when verifying the personal key

Reply:
Do we need to escape the dash as well? Is it fine if we just use the plain space character?

```suggestion
  REGEXP_STRING = "(?:#{VALID_WORD}([ -])?){#{WORD_COUNT - 1}}#{VALID_WORD}".freeze
```

Reply:
I'm not sure if the intention at the time was to allow both tabs and spaces? I'm not sure what other white space characters would be expected.

Reply:
Yeah, I wouldn't really expect that to happen, or at least pressing the <kbd>Tab</kbd> character would shift focus and not print out a tab character, but I suppose there's a bunch of different whitespace we might expect if someone's pasting it in.